kmir/src/tests/integration/data/exec-smir/references/array_elem_ref.rs

```diff
@@ -0,0 +1,20 @@
+// reference into array
+
+fn f(x: &u32) {
+  assert!(*x == 0);
+}
+
+fn g(x: *const u32) {
+//   assert!(unsafe{*x} == 0); // deref alignment check currently failing
```

Comment:
This line creates code for the pointer alignment check which currently creates thunks and gets stuck.
The alignment check (as well as the `unOpOffset`) require simulating an address according to the referenced type, its (original) size and alignment must be respected on offsets and when `transmute`ing pointers to numbers.

<img width="498" height="458" alt="image" src="https://github.com/user-attachments/assets/e83eda5d-7c5f-4d5b-90ac-8037928d0ed2" />